src/PKSim.Infrastructure/Services/NeighborhoodFinalizer.cs

```diff
@@ -20,29 +21,35 @@ public NeighborhoodFinalizer(IFlatNeighborhoodRepository flatNeighborhoodReposit
 
       public void SetNeighborsIn(Individual individual)
       {
-         foreach (var neighborhood in individual.Neighborhoods.GetAllChildren<IPKSimNeighborhood>())
-         {
-            setNeighborsIn(neighborhood, neighborhood.Name, individual.Organism);
-         }
+         individual.Neighborhoods.GetAllChildren<PKSimNeighborhood>().Each(x => setNeighborsIn(x, individual.Organism));
       }
 
-      public void SetNeighborsIn(Organism organism, IEnumerable<INeighborhoodBuilder> neighborhoods)
+      public void SetNeighborsIn(Organism organism, IReadOnlyList<NeighborhoodBuilder> neighborhoods)
       {
-         foreach (var neighborhood in neighborhoods)
-         {
-            setNeighborsIn(neighborhood, neighborhood.Name, organism);
-         }
+         neighborhoods.Each(x => setNeighborsIn(x, organism));
       }
 
-      private void setNeighborsIn(INeighborhoodBase neighborhood, string neighborhoodName, Organism organism)
+      private void setNeighborsIn(Neighborhood neighborhood, Organism organism)
       {
-         var flatNeighborhood = _flatNeighborhoodRepository.NeighborhoodFrom(neighborhoodName);
+         var flatNeighborhood = _flatNeighborhoodRepository.NeighborhoodFrom(neighborhood.Name);
 
          neighborhood.FirstNeighbor = neighborFrom(flatNeighborhood.FirstNeighborId, organism);
          neighborhood.SecondNeighbor = neighborFrom(flatNeighborhood.SecondNeighborId, organism);
       }
 
-      private IContainer neighborFrom(int neighborId, Organism organism)
+      private void setNeighborsIn(NeighborhoodBuilder neighborhood, Organism organism)
+      {
+         var flatNeighborhood = _flatNeighborhoodRepository.NeighborhoodFrom(neighborhood.Name);
+
+         neighborhood.FirstNeighborPath = neighborPathFrom(flatNeighborhood.FirstNeighborId, organism);
```

Comment:
Here this is the real change. For builder, we cannot set the container but we need to use the path instead